platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/EncryptingWrappedPersistenceManager.java

```diff
@@ -0,0 +1,105 @@
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Set;
+
+import org.apache.felix.cm.PersistenceManager;
+
+import ddf.security.encryption.EncryptionService;
+import ddf.security.encryption.impl.EncryptionServiceImpl;
+
+/**
+ * Performs encryption.
+ */
+public class EncryptingWrappedPersistenceManager extends WrappedPersistenceManager {
+
+    private static final Set<String> EXCLUDED_PROPERTIES = new HashSet<>();
+
+    static {
+        EXCLUDED_PROPERTIES.add(SERVICE_PID);
+        EXCLUDED_PROPERTIES.add(SERVICE_FACTORYPID);
+        // Not worth adding a dependency on felix's fileinstall module just for the constant
+        EXCLUDED_PROPERTIES.add("felix.fileinstall.filename");
+    }
+
+    private final EncryptionService encryptionService;
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager) {
+        super(persistenceManager);
+        this.encryptionService = new EncryptionServiceImpl();
+    }
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager,
+            EncryptionService encryptionService) {
+        super(persistenceManager);
+        this.encryptionService = encryptionService;
+    }
+
+    @Override
+    public Dictionary load(String pid) throws IOException {
+        return decryptAndReturnDictonary(super.load(pid));
+    }
+
+    @Override
+    public void store(String pid, Dictionary properties) throws IOException {
+        super.store(pid, encryptAndReturnDictonary(properties));
+    }
+
+    // The 2.3.7 version of the maven-bundle-plugin fails on lambdas,
+    // so we need to use Java 7 constructs
+    private Dictionary<String, Object> encryptAndReturnDictonary(Dictionary<String, Object> dict) {
+        Dictionary<String, Object> encryptedDictionaryToReturn = new Hashtable<>();
+        Enumeration<String> keys = dict.keys();
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+            encryptedDictionaryToReturn.put(key, encryptString(key, dict.get(key)));
+        }
+        return encryptedDictionaryToReturn;
+    }
+
+    // The 2.3.7 version of the maven-bundle-plugin fails on lambdas,
+    // so we need to use Java 7 constructs
+    private Dictionary<String, Object> decryptAndReturnDictonary(Dictionary<String, Object> dict) {
+        Dictionary<String, Object> decryptedDictionaryToReturn = new Hashtable<>();
+        Enumeration<String> keys = dict.keys();
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+            decryptedDictionaryToReturn.put(key, decryptString(key, dict.get(key)));
+        }
+        return decryptedDictionaryToReturn;
+    }
+
+    private Object encryptString(String key, Object value) {
+        if (value instanceof String && !EXCLUDED_PROPERTIES.contains(key)) {
+            return encryptionService.encrypt((String) value);
```

Comment:
❗️ The current implementation of the encryption service will return the string intact if it fails to encrypt after only having logged at debug level the issue. Is that acceptable? This means we would be potentially storing passwords un-encrypted. @stustison What is your take on that?

Reply:
It should at least be writing an audit as opposed to just a log message.